ui/app/templates/components/dashboard/vault-version-title.hbs

```diff
@@ -0,0 +1,42 @@
+<div class="has-top-margin-l has-bottom-margin-l">
+  <h1 class="title is-inline is-v-centered" data-test-dashboard-version-header>
+    {{this.versionHeader}}
+  </h1>
+  <Hds::Badge
```

Comment:
This should probably be an HDS button since it clicks to open and close a modal? Maybe a question for design, just basing that off of docs [here](https://helios.hashicorp.design/components/badge#content): 

> Since Badges are not interactive, they don’t support links. Consider moving the link outside of the Badge instead.

Reply:
@hellobontempo I updated this to be a button instead, but I still need to confirm with the design about how it will look. For right now, I've added extra styles so it looks like the current designs. 

<img width="337" alt="Screenshot 2023-07-18 at 7 45 57 AM" src="https://github.com/hashicorp/vault/assets/30884335/e7d69f3f-f846-4f6e-81dc-56a6bb48aa0c">

Reply:
Sounds good! Yeah I'm not sure what they'd want!